### PR TITLE
Refactor Batch: extract constants

### PR DESCRIPTION
**Pull request by stefano-maggiolo**
This change is [Reviewable](https://reviewable.io/reviews/cms-dev/cms/864)

### CODE REVIEW

File: cms/grading/tasktypes/Batch.py [hunk 1]
```diff
@@ -353,15 +362,14 @@ def evaluate(self, job, file_cacher):
...
                     # Check the solution with a comparator
-                    elif self.parameters[2] == "comparator":
-                        manager_filename = "checker"
-
-                        if manager_filename not in job.managers:
+                    elif self.parameters[2] == Batch.OUTPUT_EVAL_CHECKER:
+                        if Batch.CHECKER_FILENAME not in job.managers:
                             logger.error("Configuration error: missing or "
                                          "invalid comparator (it must be "
                                          "named 'checker')",
```
Comment (lerks) [inline cms/grading/tasktypes/Batch.py]: Inject constant into this message.

File: cms/grading/tasktypes/Batch.py [hunk 2]
```diff
@@ -79,14 +79,23 @@ class Batch(TaskType):
     outcome to stdout and the text to stderr.
 
     """
+    # Some local constants used in this class.
+    CORRECT_OUTPUT_FILENAME = "res.txt"
+    CHECKER_FILENAME = "checker"
+    OUTPUT_EVAL_DIFF = "diff"
+    OUTPUT_EVAL_CHECKER = "comparator"
+    COMPILATION_ALONE = "alone"
+    COMPILATION_GRADER = "grader"
```
Comment (lerks) [inline cms/grading/tasktypes/Batch.py]: There's two other hardcoded filenames, i.e., `grader.%l.` (the source) and `grader` (the executable). You could consider moving them here as well.

File: cms/grading/tasktypes/Batch.py [hunk 3]
```diff
@@ -79,14 +79,23 @@ class Batch(TaskType):
     outcome to stdout and the text to stderr.
 
     """
+    # Some local constants used in this class.
```
Comment (lerks) [inline cms/grading/tasktypes/Batch.py]: Could you split them into constants for the parameter enums and filenames?

**Comment by stefano-maggiolo** [comment]
Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions.

---

*[cms/grading/tasktypes/Batch.py, line 82 at r1](https://reviewable.io:443/reviews/cms-dev/cms/864#-L66AbpN70IWit_q9vML-r1-82:-L66I9ZNT0oc9uFJY4Iw:b-896fix) ([raw file](https://github.com/cms-dev/cms/blob/bcc0c9c06b8d00d912b3788d973156a633347800/cms/grading/tasktypes/Batch.py#L82)):*
<details><summary><i>Previously, lerks (Luca Wehrstedt) wrote…</i></summary><blockquote>

Could you split them into constants for the parameter enums and filenames?
</blockquote></details>

Done.

---

*[cms/grading/tasktypes/Batch.py, line 88 at r1](https://reviewable.io:443/reviews/cms-dev/cms/864#-L66AbpN70IWit_q9vML-r1-88:-L66IA4lYYX323bfMqhr:b-896fix) ([raw file](https://github.com/cms-dev/cms/blob/bcc0c9c06b8d00d912b3788d973156a633347800/cms/grading/tasktypes/Batch.py#L88)):*
<details><summary><i>Previously, lerks (Luca Wehrstedt) wrote…</i></summary><blockquote>

There's two other hardcoded filenames, i.e., `grader.%l.` (the source) and `grader` (the executable). You could consider moving them here as well.
</blockquote></details>

Done.

---

*[cms/grading/tasktypes/Batch.py, line 375 at r1](https://reviewable.io:443/reviews/cms-dev/cms/864#-L66AbpN70IWit_q9vML-r1-375:-L66IAnb5zrhbwfDi7Mn:b-896fix) ([raw file](https://github.com/cms-dev/cms/blob/bcc0c9c06b8d00d912b3788d973156a633347800/cms/grading/tasktypes/Batch.py#L375)):*
<details><summary><i>Previously, lerks (Luca Wehrstedt) wrote…</i></summary><blockquote>

Inject constant into this message.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cms-dev/cms/864)*

**Pull request closed without merging**